sdk/src/main/java/com/kin/ecosystem/Kin.java

```diff
@@ -74,27 +85,105 @@ private static Kin getInstance() {
+			final String networkId = kinEnvironment.getBlockchainPassphrase();
+			final String issuer = kinEnvironment.getIssuer();
+
+			KinClient kinClient = new KinClient(appContext, new ServiceProvider(networkUrl, networkId) {
+				@Override
+				protected String getIssuerAccountId() {
+					return issuer;
+				}
+			}, KIN_ECOSYSTEM_STORE_PREFIX_KEY);
+			BlockchainSourceImpl.init(eventLogger, kinClient, BlockchainSourceLocal.getInstance(appContext));
+
+			AuthRepository
+				.init(AuthLocalData.getInstance(appContext), AuthRemoteData.getInstance(instance.executorsUtil));
+
+			EventCommonDataUtil.setBaseData(appContext);
+
+			AccountManagerImpl
+				.init(AccountManagerLocal.getInstance(appContext), eventLogger, AuthRepository.getInstance(),
+					BlockchainSourceImpl.getInstance());
+
+			OrderRepository.init(BlockchainSourceImpl.getInstance(),
+				eventLogger,
+				OrderRemoteData.getInstance(instance.executorsUtil),
+				OrderLocalData.getInstance(appContext, instance.executorsUtil));
+
+			OfferRepository.init(OfferRemoteData.getInstance(instance.executorsUtil), OrderRepository.getInstance());
+
+			DeviceUtils.init(appContext);
+
+			eventLogger.send(KinSdkInitiated.create());
 		}
 	}
 
-	public static void start(@NonNull Context appContext, @NonNull String jwt, @NonNull KinEnvironment environment)
-		throws ClientException, BlockchainException {
-		if (isInstanceNull()) {
-			SignInData signInData = getJwtSignInData(jwt);
-			init(appContext, signInData, environment);
+	private static void loadDefaultsFromMetadata(Context context) throws ClientException {
+		if (context == null) {
+			return;
+		}
+
+		ApplicationInfo applicationInfo;
+		try {
+			applicationInfo = context.getPackageManager().getApplicationInfo(
+				context.getPackageName(), PackageManager.GET_META_DATA);
+		} catch (PackageManager.NameNotFoundException e) {
+			throwProvideEnvironmentMetaDataException();
+			return;
 		}
+
+		if (applicationInfo == null || applicationInfo.metaData == null) {
+			throwProvideEnvironmentMetaDataException();
+		}
+
+		Object envNameObj = applicationInfo.metaData.get(KIN_ECOSYSTEM_ENVIRONMENT_NAME_KEY);
+		if (envNameObj instanceof String) {
+			final String envName = (String) envNameObj;
+			if (Validator.isEnvironmentName(envName)) {
+				environmentName = envName;
+			} else {
+				throw new ClientException(BAD_CONFIGURATION, "Environment name: " + envName + " is not valid", null);
+			}
+		} else {
+			throwProvideEnvironmentMetaDataException();
+		}
+	}
+
+	private static void throwProvideEnvironmentMetaDataException() throws ClientException {
+		throw new ClientException(BAD_CONFIGURATION,
+			"You must provide environment meta data element in AndroidManifest.xml as a String value", null);
 	}
 
 	public static void enableLogs(final boolean enableLogs) {
 		Logger.enableLogs(enableLogs);
 	}
 
+	public static void login(@NonNull WhitelistData whitelistData, KinCallback<Void> loginCallback) {
```

Comment:
consider keeping the old `start` method as deprecated in a while, before breaking the API entirely.
(internally the start will call the new `initialize` and `login`)

Reply:
hmmm let's talk about it.

Reply:
🤔

Reply:
We decided do keep it like that, developer asked for the callback and they will not use the start without the callback.